mlir/include/mlir/Dialect/Bufferization/Transforms/Passes.td

```diff
@@ -320,6 +320,11 @@ def BufferResultsToOutParams : Pass<"buffer-results-to-out-params", "ModuleOp">
     Option<"addResultAttribute", "add-result-attr", "bool",
        /*default=*/"false",
        "Add the attribute 'bufferize.result' to all output parameters.">,
+    Option<"avoidBufferResultAllocAndCopy", "avoid-buffer-result-alloc-copy",
+       "bool", /*default=*/"false",
+       "When the returned memref has static shape and is allocated by "
```

Comment:
This description is rendered as a table (I think when running `mlir-opt -help`) and usually just one line.

How about renaming this option to `hoist-static-allocs`? And as a description for the option: `"Hoist static allocations to call sites."` (and moving the current description to the pass description above).

Reply:
Thanks for the suggestions! I have updated the code.

Reply:
This enhancement is indeed helpful to remove the overhead of result copy.
Just curious about the new name:  as the original pass always requires the the caller to allocate the buffer for the output params(newly added function argument), it might be a bit confusing that we're running with "buffer-results-to-out-params" pass but with "hoist-static-allocs=0".

Reply:
A new allocation is indeed always placed at the call site. But I would call this "hoisting" only if an equivalent allocation is removed in the body of the callee. (Hoisting = Moving) This happens only if `hoist-static-allocs=1`.

Reply:
I see, thanks for the clarification:)